### DISCUSSION TITLE
Architecture

### DISCUSSION THREAD

**Discussion by master-bogdan** [atherdon/dashb#58] [Ideas]
I think it time to discus which technologies and libs we will be use on project. Here I will write my recommendations and wishes for frontend. And in final we can make here full architecture
1. React with antd (also i think NextJS)
2. Styled components (it will be better for styling)
3. Redux state managment

@atherdon @vadim9999

**Comment by atherdon** [comment]
this is wot we have right now https://github.com/atherdon/dashb/tree/react

**Comment by vadim9999** [comment]
I am not sure about nextjs because the main requirement is make project as fast as we can. I used cra for faster setup. 
Yeah we can use styled components but what will wil do with your styles in scss? 
this project is very simple and it is not using even request to backend we havn't got it. I will be using react hooks and can use hook useReducer or createContext for state managemen, but if we will create request to api we will setup redux + redux thunk or saga.
all depends from Arthur

**Reply by master-bogdan** [reply]
@vadim9999 we can easy transform scss to styled-components

**Reply by atherdon** [reply]
I think we can think about migration into NextJS pretty soon. But yeah, right now my main goal is to create a lot of simple views, that I can show to my teammates. So we even ditched working on the database right now. Because it can eat some time.

let's think how much time it will take to migrate into NextJS and what can block us.
but probably we'll do this migration only in February or in March 2021

**Reply by atherdon** [reply]
if we make our components pretty clean, separate, and independent from everything - I think it will be easy to change things into NextJS(if it will help us to make the project even better)

**Reply by vadim9999** [reply]
Still can't understand why we need nextJS?

**Comment by master-bogdan** [comment]
@vadim9999 I want make best practices structure in project
1. Component folder will consist of component.js, index.js (this for better indexation), styles.js (here will be all components styles)
2. In components folder we can make Ui folder, where will be all typography, buttons and etc
So, if you do not mind, I can change structure to better now, and then start to make needed components, what do you think ?

**Reply by vadim9999** [reply]
Component folder contains index.js. styles.js would be great for global styling not for all components.
I am not sure aboute storing all UI in components folder it should be in src folder because we can use this styles in pages folder

**Reply by master-bogdan** [reply]
@vadim9999 for global styling, styled components have theme.js and globalstyles. In one folder with component it works like module
Yes, Ui folder will be in src/components/ui

**Reply by master-bogdan** [reply]
components folder I mean not main component folder. For example we have component Button, so folder will be consist of Button.js, index.js, styles,js. All components should have this architecture

**Reply by master-bogdan** [reply]
@vadim9999 it will be structure like in Mardown project in next js frontend

**Reply by vadim9999** [reply]
Ok lets start to do then I will see changes

**Reply by master-bogdan** [reply]
@vadim9999 ok, today, tomorrow I will make ESLINT, prettier settings, change structure and then will be work on new components

**Comment by master-bogdan** [comment]
@vadim9999 where is the package.json ?

**Reply by vadim9999** [reply]
@master-bogdan  Sorry, the command npx create-react-app added package.json to gitignore. I have just fixed it

**Comment by vadim9999** [comment]
@master-bogdan please remove empty styles.js files and index.js (maybe it's greate for indexing but it's a lot of unnecessary files)
![image](https://user-images.githubusercontent.com/35640573/104849755-7dc50980-58fc-11eb-9b20-2e9f752fda9d.png)

**Comment by master-bogdan** [comment]
about empty styles.js, I agree
About index.js it from best practices think, to not have import like 
`import Table from 'components/Table/Table';`

**Reply by vadim9999** [reply]
@master-bogdan  I don't like idea with index.js. Because it's stores only two rows of code and no more. 
import Table from 'components/Table/Table'; this import looks better than two files and more clear

**Reply by master-bogdan** [reply]
Ok, as you wish

**Reply by vadim9999** [reply]
Thank you!

**Comment by vadim9999** [comment]
![image](https://user-images.githubusercontent.com/35640573/104850191-acdc7a80-58fe-11eb-8664-3a99d63192f5.png)

a lot of warnings are you working on linux windows or mac?

**Reply by master-bogdan** [reply]
on Linux

**Reply by vadim9999** [reply]
At the moment I am on windows. I have just fixed this issue adding in eslint a rule

**Reply by master-bogdan** [reply]
@vadim9999 i think this rule must be in .prettierrc, in main file only link to prettier settings

**Reply by vadim9999** [reply]
I have found this solution on stackoverflow. If you have better idea, please apply it

**Comment by vadim9999** [comment]
@master-bogdan Why have you deleted index.js from components?
![image](https://user-images.githubusercontent.com/35640573/104851128-20cd5180-5904-11eb-9175-74be26ecb45b.png)

**Reply by master-bogdan** [reply]
It was when I did reorganisation. If you want, I can get it back

**Reply by vadim9999** [reply]
Yes, please change it back

**Comment by master-bogdan** [comment]
@vadim9999 please, give me next tasks to do

**Comment by vadim9999** [comment]
@master-bogdan you have two tasks in progress. Please open kanban board in github. Tab Projects
